<a href="https://colab.research.google.com/github/PatilTruptiU/EDA-PROJECT---Play-store-Analysis-/blob/main/play_store_data_and_user_reviews_analysis_EDA_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##Let us begin with importing the libraries

In [ ]:
#importing numpy package
import numpy as np
# Import Numpy Package
import pandas as pd 
# Import matplotlib
import matplotlib.pyplot as plt
# Import seaborn
import seaborn as sns

## mounting the drive 

In [ ]:
 from google.colab import drive 
 drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
# Loading csv files
file_path1='/content/drive/MyDrive/Play Store Data.csv'
file_path2='/content/drive/MyDrive/User Reviews.csv'

In [ ]:
#Loading the dataset as a pandas dataframe
play_store_df = pd.read_csv(file_path1)
user_review_df = pd.read_csv(file_path2)

In [ ]:
p_df1 = play_store_df
u_df2 = user_review_df

##**1. Plat store data** 
##data information

In [ ]:
#firstly check shape of  dataframe 
p_df1.shape

(10841, 13)

In [ ]:
# columns info
p_df1.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

In [ ]:
p_df1.columns.str.replace(' ','_')

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content_Rating', 'Genres', 'Last_Updated', 'Current_Ver',
       'Android_Ver'],
      dtype='object')

### **note -** 
Given play store dataset has 10841 rows and 13 columns 
  
###**columns information :**

App : The name of the app

Category : The category of the app

Rating : The rating of the app in the Play Store

Reviews : The number of reviews of the app

Size : The size of the app

Install : The number of installs of the app

Type : The type of the app (Free/Paid)

The price of the app (0 if it is Free)

Content Rating :The appropiate target audience of the app

Genres: The genre of the app

Last Updated : The date when the app was last updated

Current Ver : The current version of the app

Android Ver : The minimum Android version required to run the app





In [ ]:
#head or the first 10 rows of the dataframe
p_df1.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [ ]:
p_df1.tail()

App             Category  \
10836                               Sya9a Maroc - FR               FAMILY   
10837               Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                         Parkinson Exercices FR              MEDICAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating             Genres      Last Updated         Current Ver  \
10836       Everyone          Education     July 25, 2017                1.48   
10837       Everyone          Education      July 6, 2018                 1.0   
10838       Everyone            Medical  January 20, 2017                 1.0   
10839     Mature 17+  Books & Reference  January 19, 2015  Varies with device   
10840       Everyone          Lifestyle     July 25, 2018  Varies with device   

              Android Ver  
10836          4.1 and up  
10837          4.1 and up  
10838          2.2 and up  
10839  Varies with device  
10840  Varies with device

In [ ]:
# info method - To find out the data type & non-null count of each column 
p_df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


##**Cleaning data**

In [ ]:
#replacing + by '' and ','by ''from Prices column
Installing = [ i for i in p_df1['Installs']]
def clean_Installing(Installing_list):
  cleaned_data = []
  for Installs in Installing_list:
    if '+' in Installs:
      Installs = Installs.replace('+','')
    if ',' in Installs:
      Installs = Installs.replace(',','')
    cleaned_data.append(Installs)
  return cleaned_data


In [ ]:
p_df1['Installs'] = clean_Installing(Installing)
p_df1.Installs.unique()

array(['10000', '500000', '5000000', '50000000', '100000', '50000',
       '1000000', '10000000', '5000', '100000000', '1000000000', '1000',
       '500000000', '50', '100', '500', '10', '1', '5', '0', 'Free'],
      dtype=object)

In [ ]:
p_df1.loc[p_df1['Installs']=='Free']


App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  Rating\t  
10472       1.0.19  4.0 and up         NaN      19.0

here in above row seems that due to the categaory value missing all values are shifted one plce to left.so it's better that we will drop this row from datafram

In [ ]:
p_df1.drop([10472],axis=0,inplace=True)

In [ ]:
p_df1.shape

(10840, 14)

In [ ]:
#Droping null values from the dataset
p_df1.dropna()

App             Category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10834                                      FR Calculator               FAMILY   
10836                                   Sya9a Maroc - FR               FAMILY   
10837                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating   Reviews                Size    Installs  Type Price  \
0         4.1     159.0                 19M     10000.0  Free     0   
1         3.9     967.0                 14M    500000.0  Free     0   
2         4.7   87510.0                8.7M   5000000.0  Free     0   
3         4.5  215644.0                 25M  50000000.0  Free     0   
4         4.3     967.0                2.8M    100000.0  Free     0   
...       ...       ...                 ...         ...   ...   ...   
10834     4.0       7.0                2.6M       500.0  Free     0   
10836     4.5      38.0                 53M      5000.0  Free     0   
10837     5.0       4.0                3.6M       100.0  Free     0   
10839     4.5     114.0  Varies with device      1000.0  Free     0   
10840     4.5  398307.0                 19M  10000000.0  Free     0   

      Content Rating                     Genres      Last Updated  \
0           Everyone               Art & Design   January 7, 2018   
1           Everyone  Art & Design;Pretend Play  January 15, 2018   
2           Everyone               Art & Design    August 1, 2018   
3               Teen               Art & Design      June 8, 2018   
4           Everyone    Art & Design;Creativity     June 20, 2018   
...              ...                        ...               ...   
10834       Everyone                  Education     June 18, 2017   
10836       Everyone                  Education     July 25, 2017   
10837       Everyone                  Education      July 6, 2018   
10839     Mature 17+          Books & Reference  January 19, 2015   
10840       Everyone                  Lifestyle     July 25, 2018   

              Current Ver         Android Ver  Rating\t  
0                   1.0.0        4.0.3 and up       4.1  
1                   2.0.0        4.0.3 and up       3.9  
2                   1.2.4        4.0.3 and up       4.7  
3      Varies with device          4.2 and up       4.5  
4                     1.1          4.4 and up       4.3  
...                   ...                 ...       ...  
10834               1.0.0          4.1 and up       4.0  
10836                1.48          4.1 and up       4.5  
10837                 1.0          4.1 and up       5.0  
10839  Varies with device  Varies with device       4.5  
10840  Varies with device  Varies with device       4.5  

[9360 rows x 14 columns]

here in given data last three colums are not much usefull so we drop it
  

In [ ]:
#let us drop three coumns which are not much useful
p_df= p_df1.drop(['Last Updated','Current Ver','Android Ver'], axis=1)

App             Category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10836                                   Sya9a Maroc - FR               FAMILY   
10837                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                             Parkinson Exercices FR              MEDICAL   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating   Reviews                Size    Installs  Type Price  \
0         4.1     159.0                 19M     10000.0  Free     0   
1         3.9     967.0                 14M    500000.0  Free     0   
2         4.7   87510.0                8.7M   5000000.0  Free     0   
3         4.5  215644.0                 25M  50000000.0  Free     0   
4         4.3     967.0                2.8M    100000.0  Free     0   
...       ...       ...                 ...         ...   ...   ...   
10836     4.5      38.0                 53M      5000.0  Free     0   
10837     5.0       4.0                3.6M       100.0  Free     0   
10838     NaN       3.0                9.5M      1000.0  Free     0   
10839     4.5     114.0  Varies with device      1000.0  Free     0   
10840     4.5  398307.0                 19M  10000000.0  Free     0   

      Content Rating                     Genres  Rating\t  
0           Everyone               Art & Design       4.1  
1           Everyone  Art & Design;Pretend Play       3.9  
2           Everyone               Art & Design       4.7  
3               Teen               Art & Design       4.5  
4           Everyone    Art & Design;Creativity       4.3  
...              ...                        ...       ...  
10836       Everyone                  Education       4.5  
10837       Everyone                  Education       5.0  
10838       Everyone                    Medical       NaN  
10839     Mature 17+          Books & Reference       4.5  
10840       Everyone                  Lifestyle       4.5  

[10840 rows x 11 columns]

In [ ]:
p_df.shape

(9360, 11)

In [ ]:
#replacing $ by '' from Prices column
prices = [i for i in p_df['Price']]

def clean_prices(prices_list):
    cleaned_data = []
    for price in prices_list:
        if '$' in price:
            price = price.replace('$', '')
        cleaned_data.append(price)
    return cleaned_data

In [ ]:
p_df['Price'] = clean_prices(prices)
p_df.Price.unique()

array(['0', '4.99', '3.99', '6.99', '7.99', '5.99', '2.99', '3.49',
       '1.99', '9.99', '7.49', '0.99', '9.00', '5.49', '10.00', '24.99',
       '11.99', '79.99', '16.99', '14.99', '29.99', '12.99', '2.49',
       '10.99', '1.50', '19.99', '15.99', '33.99', '39.99', '3.95',
       '4.49', '1.70', '8.99', '1.49', '3.88', '399.99', '17.99',
       '400.00', '3.02', '1.76', '4.84', '4.77', '1.61', '2.50', '1.59',
       '6.49', '1.29', '299.99', '379.99', '37.99', '18.99', '389.99',
       '8.49', '1.75', '14.00', '2.00', '3.08', '2.59', '19.40', '3.90',
       '4.59', '15.46', '3.04', '13.99', '4.29', '3.28', '4.60', '1.00',
       '2.95', '2.90', '1.97', '2.56', '1.20'], dtype=object)

In [ ]:
# Converting KB to MB form size columns 
p_df['Size'] = p_df['Size'].apply(lambda x: str(x).replace('Varies with device', 'NaN') if 'Varies with device' in str(x) else x)
p_df['Size'] = p_df['Size'].apply(lambda x: str(x).replace('M', '') if 'M' in str(x) else x)
p_df['Size'] = p_df['Size'].apply(lambda x: str(x).replace(',', '') if 'M' in str(x) else x)
p_df['Size'] = p_df['Size'].apply(lambda x: float(str(x).replace('k', '')) / 1000 if 'k' in str(x) else x)
p_df['Size'] = p_df['Size'].apply(lambda x: float(x))

In [ ]:
p_df.head(10)

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   
5                         Paper flowers instructions  ART_AND_DESIGN     4.4   
6            Smoke Effect Photo Maker - Smoke Editor  ART_AND_DESIGN     3.8   
7                                   Infinite Painter  ART_AND_DESIGN     4.1   
8                               Garden Coloring Book  ART_AND_DESIGN     4.4   
9                      Kids Paint Free - Drawing Fun  ART_AND_DESIGN     4.7   

    Reviews  Size    Installs  Type  Price Content Rating  \
0     159.0  19.0     10000.0  Free    0.0       Everyone   
1     967.0  14.0    500000.0  Free    0.0       Everyone   
2   87510.0   8.7   5000000.0  Free    0.0       Everyone   
3  215644.0  25.0  50000000.0  Free    0.0           Teen   
4     967.0   2.8    100000.0  Free    0.0       Everyone   
5     167.0   5.6     50000.0  Free    0.0       Everyone   
6     178.0  19.0     50000.0  Free    0.0       Everyone   
7   36815.0  29.0   1000000.0  Free    0.0       Everyone   
8   13791.0  33.0   1000000.0  Free    0.0       Everyone   
9     121.0   3.1     10000.0  Free    0.0       Everyone   

                      Genres  Rating\t  
0               Art & Design       4.1  
1  Art & Design;Pretend Play       3.9  
2               Art & Design       4.7  
3               Art & Design       4.5  
4    Art & Design;Creativity       4.3  
5               Art & Design       4.4  
6               Art & Design       3.8  
7               Art & Design       4.1  
8               Art & Design       4.4  
9    Art & Design;Creativity       4.7

In [ ]:
#checking all values unique values of Category column
p_df.Category.unique()

array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION'],
      dtype=object)

In [ ]:
p_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9360 entries, 0 to 10840
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             9360 non-null   object 
 1   Category        9360 non-null   object 
 2   Rating          9360 non-null   float64
 3   Reviews         9360 non-null   float64
 4   Size            9360 non-null   object 
 5   Installs        9360 non-null   float64
 6   Type            9360 non-null   object 
 7   Price           9360 non-null   object 
 8   Content Rating  9360 non-null   object 
 9   Genres          9360 non-null   object 
 10  Rating	         9360 non-null   float64
dtypes: float64(4), object(7)
memory usage: 877.5+ KB


here many datatypes are object we will convert it into respective appropriate dtype

In [ ]:
#change the datatype
p_df['Installs'] = p_df['Installs'].astype(float)
p_df['Reviews'] = p_df['Reviews'].astype(float)
p_df['Price'] = p_df['Price'].astype(float)
p_df['Size'] = p_df['Size'].astype(float)
p_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9360 entries, 0 to 10840
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             9360 non-null   object 
 1   Category        9360 non-null   object 
 2   Rating          9360 non-null   float64
 3   Reviews         9360 non-null   float64
 4   Size            7723 non-null   float64
 5   Installs        9360 non-null   float64
 6   Type            9360 non-null   object 
 7   Price           9360 non-null   float64
 8   Content Rating  9360 non-null   object 
 9   Genres          9360 non-null   object 
 10  Rating	         9360 non-null   float64
dtypes: float64(6), object(5)
memory usage: 877.5+ KB


In [ ]:
#checking all unique values & count of Type columns
p_df1.Type.unique()
p_df1['Type'].value_counts()

array(['Free', 'Paid', nan], dtype=object)

In [ ]:
#checking all unique values & count of Type columns
p_df['Content Rating'].unique()
p_df['Content Rating'].value_counts()

Everyone           7414
Teen               1084
Mature 17+          461
Everyone 10+        397
Adults only 18+       3
Unrated               1
Name: Content Rating, dtype: int64

In [ ]:
#checking all unique values & count of Genres columns
p_df['Genres'].unique()
p_df['Genres'].value_counts()

Tools                      732
Entertainment              533
Education                  468
Action                     358
Productivity               351
                          ... 
Parenting;Brain Games        1
Card;Brain Games             1
Tools;Education              1
Entertainment;Education      1
Strategy;Creativity          1
Name: Genres, Length: 115, dtype: int64

In [ ]:
#checking all unique values of Category columns
p_df['Category'].unique()

array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION'],
      dtype=object)

In [ ]:
#describe method - It shows statistical summer of columns that has numeric data 
p_df.describe() 

Rating       Reviews      Installs        Price     Rating\t
count  9360.000000  9.360000e+03  9.360000e+03  9360.000000  9360.000000
mean      4.191838  5.143767e+05  1.790875e+07     0.961279     4.191838
std       0.515263  3.145023e+06  9.126637e+07    15.821640     0.515263
min       1.000000  1.000000e+00  1.000000e+00     0.000000     1.000000
25%       4.000000  1.867500e+02  1.000000e+04     0.000000     4.000000
50%       4.300000  5.955000e+03  5.000000e+05     0.000000     4.300000
75%       4.500000  8.162750e+04  5.000000e+06     0.000000     4.500000
max       5.000000  7.815831e+07  1.000000e+09   400.000000     5.000000

In [ ]:
#letus check the shape after dropping which we use for futher analysis
p_df.shape

(9360, 11)

In [ ]:
p_df.isnull().sum()
#this shows how many null values after dropna method

App               0
Category          0
Rating            0
Reviews           0
Size              0
Installs          0
Type              0
Price             0
Content Rating    0
Genres            0
Rating\t          0
dtype: int64

In [ ]:
def change_int(x):
  x=x.replace(',','')
  x=x.replace('+','')
  x=x.replace('M','')
  x=x.replace('K','')
  x=x.replace('$','')
  return x

In [ ]:
#p_df['Size']=p_df['Size'].apply(change)
#p_df['Installs']=p_df['Installs'].apply(change_int)
p_df['Reviews']=p_df['Reviews'].apply(change_int)